src/Npgsql/Internal/Converters/Temporal/DateTimeConverterResolvers.cs

```diff
@@ -32,19 +32,17 @@ public override PgConverterResolution Get(DateTime value, PgTypeId? expectedPgTy
     {
         if (value.Kind is DateTimeKind.Utc)
         {
-            if (expectedPgTypeId == _timestamp)
-            {
-                throw new ArgumentException(
+            // We coalesce with expectedPgTypeId to throw on unknown type ids.
+            return expectedPgTypeId == _timestamp
+                ? throw new ArgumentException(
                     "Cannot write DateTime with Kind=UTC to PostgreSQL type 'timestamp without time zone', " +
                     "consider using 'timestamp with time zone'. " +
-                    "Note that it's not possible to mix DateTimes with different Kinds in an array/range.", nameof(value));
-            }
-
-            // We coalesce with expectedPgTypeId to throw on unknown type ids.
-            return GetDefault(expectedPgTypeId ?? _timestampTz);
+                    "Note that it's not possible to mix DateTimes with different Kinds in an array/range.", nameof(value))
+                : GetDefault(expectedPgTypeId ?? _timestampTz);
         }
 
-        if (expectedPgTypeId == _timestampTz)
+        if (expectedPgTypeId == _timestampTz
+            && !(_dateTimeInfinityConversions && (value == DateTime.MinValue || value == DateTime.MaxValue)))
```

Comment:
Probably want this check on the values with its own exception at the start? (it's missing on the utc side atm)

Reply:
I think DateTime.{Min,Max}Value are always Unspecified no?

In any case, I'll go ahead and merge, we can fix later (and hopefully if something's missing we should see it through test failures).

Reply:
Those specific values are, yes, however doing `DateTime.MaxValue.ToUniversalTime()` should still IMO work. Just as we convert for min/max instants in nodatime.